### PR TITLE
make: adapt develhelp for all examples

### PR DESCRIPTION
**Pull request by smlng**
factored out from #8029, disable develhelp for all examples

### CODE REVIEW

**Comment by miri64** [comment]
Please remember to wait with this, until #7925 is merged.

**Comment by kaspar030** [comment]
> disable develhelp for all examples

Why? This would disable all assertions, ps, extra checks, ...

**Comment by smlng** [comment]
@kaspar030 IMHO the examples are more like _productive_ applications, hence DEVELHELP would be off by default, but can still be enabled (e.g. in CI) when needed. On the other hand, for `tests` as in #8080 DEVELHELP is on by default, because those are tests and assertions etc should be checked.

**Comment by kaspar030** [comment]
> IMHO the examples are more like productive applications, hence DEVELHELP would be off by default

Why? What's there to gain?

**Comment by smlng** [comment]
> Why? What's there to gain?

Could ask the same the other way around 😉 Or: what exactly is *lost* by disabling DEVELHELP for examples, it can be enabled (as said) easily e.g. for CI builds and any prevalent issues should be discovered/pop up compiling/running the tests not examples.

If the examples (should) target RIOT beginners and users (not only developers), I don't see the point of having DEVELHELP enabled by default. It increased binary sizes and may through rather pedantic compiler errors (not necessarily causing any runtime issues) which only irritate inexperienced users and frustrate them right at the start.

**Comment by bergzand** [comment]
First thing that comes to mind is the shorted output of the `ps` command. I think that the expanded output including the process names gives some valuable insight in the threading of RIOT for newcomers. I dont know if it can be enabled by a separate flag, but I think that is one of the valuable "features" of enabling DEVELHELP.

**Comment by smlng** [comment]
> First thing that comes to mind is the shorted output of the ps command.

Agreed, but then IMHO it would be better to change/enhance the default output of the `ps` command, for instance by having thread names - while something like stack size and so on is still only available with DEVELHELP.

**Comment by smlng** [comment]
Okay, to ease things (for now) I let DEVELHELP state unchanged for examples, i.e., leave as is: on for some off for other examples. We may change this later on. 

Regarding `ps` output: what good is this command without thread names to anyone (user or developer)? Or what's its purpose for non-developers and its limited output (without DEVELHELP) anyway? But that is another discussion, unrelated here.

TL;DR: I revert my changes, and leave state of DEVELHELP (on/off) for examples as is now.

**Comment by miri64** [comment]
> Regarding ps output: what good is this command without thread names to anyone (user or developer)? Or what's its purpose for non-developers and its limited output (without DEVELHELP) anyway? But that is another discussion, unrelated here.

Seeing what state a thread is in. Very helpful if you run into a deadlock ;-).

**Comment by miri64** [comment]
Also not having no names doesn't mean you can't identify which thread is which (there is still the PID). It's just *easier* with names.

File: examples/nanocoap_server/Makefile [hunk 1]
```diff
@@ -35,7 +35,7 @@ USEMODULE += shell_commands
 # Comment this out to disable code in RIOT that does safety checking
 # which is not needed in a production environment but helps in the
 # development process:
-DEVELHELP ?= 1
+DEVELHELP ?= 0
```
Comment (kaspar030) [inline examples/nanocoap_server/Makefile]: Nitpick: the comment doesn't make sense if the line says ```?=0```. Feel free to change the line to ```#DEVELHELP ?= 1```

**Comment by gebart** [comment]
I agree with Martine and Kaspar, for newcomers the DEVELHELP switch should be on by default.

Regarding ps without DEVELHELP: 
I sometimes call the ps function directly from GDB when I run into a hardfault or other bad situation, it's better to get some output without thread names than nothing, though it was more useful before the RIOT support was updated for openocd (http://openocd.zylin.com/#/c/4256/, please try it and add your code reviews to their tracker so it can be merged into master!), now the `info threads` command provides some of the same information.

**Comment by smlng** [comment]
rebased and squashed

File: examples/nanocoap_server/Makefile [hunk 2]
```diff
@@ -35,7 +35,7 @@ USEMODULE += shell_commands
 # Comment this out to disable code in RIOT that does safety checking
 # which is not needed in a production environment but helps in the
 # development process:
-#CFLAGS += -DDEVELHELP
+DEVELHELP ?= 0
```
Comment (cladmi) [inline examples/nanocoap_server/Makefile]: Only problem here is the same as kaspaar, but it is not shown as he already ACKed

> Nitpick: the comment doesn't make sense if the line says ?=0. Feel free to change the line to #DEVELHELP ?= 1

Comment (smlng) [reply examples/nanocoap_server/Makefile]: yeah right, missed to adapt that one

**Comment by smlng** [comment]
@cladmi adapted comment

**Comment by cladmi** [comment]
Fixing and amending commit in the same one, its harder to read… (just kidding :D)
